<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Scan_Code_Dlg</class>
 <widget class="QDialog" name="Scan_Code_Dlg">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>910</width>
    <height>588</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>910</width>
    <height>588</height>
   </size>
  </property>
  <property name="windowTitle">
   <string notr="true">сканирование штрих кода</string>
  </property>
  <property name="sizeGripEnabled">
   <bool>true</bool>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_6">
     <property name="spacing">
      <number>19</number>
     </property>
     <item>
      <widget class="QFrame" name="dialogTopFrame">
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
       <property name="frameShape">
        <enum>QFrame::Box</enum>
       </property>
       <layout class="QVBoxLayout" name="vl" stretch="0,0,0,0,0">
        <item>
         <widget class="QLabel" name="label">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="text">
           <string>просканируйте штрих код маркировки (сканером в режиме клавиатуры)</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
          <property name="textInteractionFlags">
           <set>Qt::NoTextInteraction</set>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,0">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="DatamatrixTextEdit" name="plte_datamatrix">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>10</pointsize>
               </font>
              </property>
              <property name="plainText">
               <string notr="true"/>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <property name="leftMargin">
               <number>11</number>
              </property>
              <property name="topMargin">
               <number>11</number>
              </property>
              <property name="rightMargin">
               <number>11</number>
              </property>
              <item>
               <widget class="QFrame" name="frame">
                <property name="frameShape">
                 <enum>QFrame::Box</enum>
                </property>
                <property name="midLineWidth">
                 <number>1</number>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_8">
                 <item>
                  <widget class="QGroupBox" name="groupBox">
                   <property name="title">
                    <string notr="true">символом GS считать :</string>
                   </property>
                   <property name="flat">
                    <bool>false</bool>
                   </property>
                   <layout class="QVBoxLayout" name="verticalLayout_5">
                    <item>
                     <widget class="QCheckBox" name="chk_auto_detection">
                      <property name="text">
                       <string>определить автоматически</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="CustomColorPlaceholderLineEdit" name="ledt_redefineGS">
                      <property name="minimumSize">
                       <size>
                        <width>0</width>
                        <height>0</height>
                       </size>
                      </property>
                      <property name="maximumSize">
                       <size>
                        <width>1644465</width>
                        <height>16777215</height>
                       </size>
                      </property>
                      <property name="font">
                       <font>
                        <pointsize>10</pointsize>
                       </font>
                      </property>
                      <property name="toolTip">
                       <string notr="true">&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:7.5pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Вы можете посмотреть в логе как сканер передает GS символ, &lt;br /&gt;Например он передает его как F8  (рядом его значение 0х42)&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Указываем здесь 0x42 и программа заменит все F8 (0x42) на GS (0x1d)&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Если надо 2 сканкода написать можно через запятую, например&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;0x38,0x51 это Alt PgDown будет заменяться на один байт GS (0x1d).&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Сканер может передавать и такое 0x152 (это rклавиша Ins),&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;тогда указываем &lt;/span&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;0x152 (три цифры)&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                      </property>
                      <property name="inputMask">
                       <string notr="true"/>
                      </property>
                      <property name="text">
                       <string notr="true"/>
                      </property>
                      <property name="maxLength">
                       <number>100</number>
                      </property>
                      <property name="placeholderText">
                       <string notr="true">0x152,0x42</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QPushButton" name="btn_save_redifinedGS">
                      <property name="text">
                       <string>сохранить</string>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <property name="leftMargin">
             <number>11</number>
            </property>
            <property name="topMargin">
             <number>11</number>
            </property>
            <item>
             <widget class="QPushButton" name="btn_clear">
              <property name="text">
               <string>очистить</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btn_show_log">
              <property name="text">
               <string>показать лог</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="chk_enable_log">
              <property name="text">
               <string notr="true">вести лог</string>
              </property>
              <property name="checked">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="chk_clearAfterScanned">
              <property name="text">
               <string notr="true">очищать окно
после сканирования</string>
              </property>
              <property name="shortcut">
               <string notr="true"/>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="chk_closeWin_afterEnter">
              <property name="text">
               <string notr="true">закрывать окно после
 прихода Enter</string>
              </property>
              <property name="shortcut">
               <string notr="true"/>
              </property>
              <property name="checked">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="chk_showBase64">
              <property name="text">
               <string notr="true">показывать base64</string>
              </property>
              <property name="shortcut">
               <string notr="true"/>
              </property>
              <property name="checked">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btn_show_app_log">
              <property name="text">
               <string>лог приложения</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QFrame" name="fr_showBase64">
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>base64 : </string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPlainTextEdit" name="plte_base64">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>50</height>
              </size>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QLabel" name="lbl_warning">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="btn_cancel">
            <property name="text">
             <string>отменить</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="btn_accept">
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="text">
             <string>принять</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DatamatrixTextEdit</class>
   <extends>QPlainTextEdit</extends>
   <header location="global">kb_scanner/kb_wgt/datamatrix_text_edit.h</header>
  </customwidget>
  <customwidget>
   <class>CustomColorPlaceholderLineEdit</class>
   <extends>QLineEdit</extends>
   <header>customcolorplaceholderlineedit.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
